### PR TITLE
P2-814 Implement appender according to design

### PR DESCRIPTION
**Pull request by hansjovis**
## Summary

This PR can be summarized in the following changelog entry:

* [@yoast/schema-blocks] Fixes the styling of the block appender used in the `InnerBlocks` block instruction. 

## Relevant technical choices:

* 

## Test instructions

This PR can be tested by following these steps:

* Create a post.
* Add a job posting block.
* Select the job posting block (if it is not selected already).
* You should see a block appender at the bottom of the job posting.
	* It should look like [the block appender in the design](https://www.sketch.com/s/ab6ebb5c-a390-434c-bdf1-8e3240cadcef/a/e7aAGz).
	  Screenshot 2021-04-13 at 11 22 38
	* **Note**: It may look slightly different based on the theme you have installed on your website. E.g. a different background and/or font. All in all the styling should look consistent with the rest of the theme. 
* Make sure that the block appender works as expected. 
	* E.g. remove the job description from the job posting and re-add it via the block appender.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  * The location and base salary inner blocks of the job posting, which have inner blocks to be able to add variations (e.g. remote location or office location), should **not** have an appender.

## UI changes
* [x] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Quality assurance

* [x] I have tested this code to the best of my abilities
* [x] I have added unittests to verify the code works as intended

Fixes #

### CODE REVIEW

File: packages/schema-blocks/css/schema-blocks.css [hunk 1]
```diff
@@ -50,6 +50,10 @@
 	color: #0073aa;
 }
 
+.yoast-schema-block-appender__content {
```
Comment (increddibelly) [inline packages/schema-blocks/css/schema-blocks.css:53]: if the goal is to make a component look consistent with other components - why do we need *new* styling?

File: packages/schema-blocks/src/functions/presenters/BlockAppender.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { createElement } from "@wordpress/element";
+import { Inserter } from "@wordpress/block-editor";
+
+/**
+ * Properties for the BlockAppender.
+ */
+export interface BlockAppenderProps {
```
Comment (increddibelly) [inline packages/schema-blocks/src/functions/presenters/BlockAppender.tsx:7]: not used. maybe less is more this time?

File: packages/schema-blocks/src/functions/presenters/BlockAppender.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+ * Block Appender for adding an inner block to a schema block.
+ *
+ * @param clientId The id of the parent block in which a block should be added when using the appender.
+ * @param label A string
+ *
+ * @constructor
+ */
+export default function BlockAppender( { clientId, label }: BlockAppenderProps ) {
```
Comment (increddibelly) [inline packages/schema-blocks/src/functions/presenters/BlockAppender.tsx:20]:
```suggestion
export default function BlockAppender( clientId: string, label: string ) {
```

File: packages/schema-blocks/src/functions/presenters/BlockAppender.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * Properties for the BlockAppender.
+ */
+export interface BlockAppenderProps {
+	clientId: string;
+	label: string;
+}
+
+/**
+ * Block Appender for adding an inner block to a schema block.
```
Comment (increddibelly) [inline packages/schema-blocks/src/functions/presenters/BlockAppender.tsx:13]:
```suggestion
 * Block Appender for adding a new block to an innerblocks block.
```

File: packages/schema-blocks/src/functions/presenters/BlockAppender.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+		<div
+			data-root-client-id={ clientId }
+			className="wp-block block-editor-default-block-appender"
+		>
+			<p className="block-editor-default-block-appender__content yoast-schema-block-appender__content">
+				{ label }
+			</p>
+			<Inserter
+				rootClientId={ clientId }
+				position="bottom right"
```
Comment (increddibelly) [inline packages/schema-blocks/src/functions/presenters/BlockAppender.tsx:31]: bottom? is that correct?

File: packages/schema-blocks/src/functions/presenters/BlockAppender.tsx (hunk 3 above)
Comment (hansjovis) [reply packages/schema-blocks/src/functions/presenters/BlockAppender.tsx:20]: We can implement this change, but that would mean that we can't use this function as a React component. That would also mean that we can't use JSX to instantiate this component.

[React components are functions (or classes) that get a single `props` object as its argument.](https://reactjs.org/docs/components-and-props.html#function-and-class-components)

File: packages/schema-blocks/src/functions/presenters/BlockAppender.tsx (hunk 5 above)
Comment (hansjovis) [reply packages/schema-blocks/src/functions/presenters/BlockAppender.tsx:31]: I copied it from the `DefaultBlockAppender`. 

I guess if the space for the 'Add a job to your job posting' label is small enough that its text wraps multiple lines, we want the block appender to be at the bottom right?

File: packages/schema-blocks/src/functions/presenters/BlockAppender.tsx (hunk 2 above)
Comment (hansjovis) [reply packages/schema-blocks/src/functions/presenters/BlockAppender.tsx:7]: See [my other comment](https://github.com/Yoast/javascript/pull/1155/files#r612361618) 😉 .

Unfortunately, if we want to use the `BlockAppender` as a React component, we need to have a single `props` argument to the function. And TypeScript expects that each argument has an appropriate type.

File: packages/schema-blocks/css/schema-blocks.css (hunk 1 above)
Comment (hansjovis) [reply packages/schema-blocks/css/schema-blocks.css:53]: Without this CSS the styling looks inconsistent with the other components (e.g. the 'Start writing or type / to choose a block' label). Without it, the label has 100% opacity.

But you are right that we should not ad any styling if absolutely necessary. I will look if I can add a class from WordPress so it gets the styling it needs from WordPress.

Comment (hansjovis) [reply packages/schema-blocks/css/schema-blocks.css:53]: I looked into it. No class that I added solves the styling issue 😞.

It is really weird. If I use the browser inspector to inspect the WordPress elements from which I want to copy the styling, the inspector says that its color is `rgb(30, 30, 30)`, but that does not correspond with how the elements look inside the editor.

File: packages/schema-blocks/src/functions/presenters/BlockAppender.tsx (hunk 2 above)
Comment (increddibelly) [reply packages/schema-blocks/src/functions/presenters/BlockAppender.tsx:7]: awesome! thanks.

File: packages/schema-blocks/src/functions/presenters/BlockAppender.tsx (hunk 3 above)
Comment (increddibelly) [reply packages/schema-blocks/src/functions/presenters/BlockAppender.tsx:20]: that's great. thanks

File: packages/schema-blocks/css/schema-blocks.css (hunk 1 above)
Comment (increddibelly) [reply packages/schema-blocks/css/schema-blocks.css:53]: we're using a data-label for the label. could it be better with a `<P>` ?

**Review by increddibelly — APPROVED** [review]

**Comment by johannadevos** [comment]
Acceptance: ✅ 

- [x] The appender becomes visible when the Job Posting block is selected.
- [x] It is possible to add a block through the appender.
- [x] The styling looks good.
- [x] The variation blocks don't have an appender.
- [ ] It is a known issue that clicking on appender selects the parent block: https://yoast.atlassian.net/browse/P2-782.

**Pull request merged**